### ISSUE TITLE
ActivityLogs DefineQuery Bad Request

### ISSUE THREAD

**Issue by pppchris** [Azure/azure-libraries-for-net#511]
```
var log = await Repo.ActivityLogs
                .DefineQuery()
                .StartingFrom(DateTime.Now.AddDays(-7))
                .EndsBefore(DateTime.Now)
                .WithAllPropertiesInResponse()
                .FilterByResource(id)
               .ExecuteAsync();
```

Operation returned an invalid status code 'BadRequest'
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

Exception Details: Microsoft.Azure.Management.Monitor.Fluent.Models.ErrorResponseException: Operation returned an invalid status code 'BadRequest'

```
[ErrorResponseException: Operation returned an invalid status code 'BadRequest']
   Microsoft.Azure.Management.Monitor.Fluent.<ListWithHttpMessagesAsync>d__5.MoveNext() +4302
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +60
   Microsoft.Azure.Management.Monitor.Fluent.<ListAsync>d__0.MoveNext() +354
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +60
   Microsoft.Azure.Management.Monitor.Fluent.<<ListEventDataAsync>b__0>d.MoveNext() +288
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +60
   Microsoft.Azure.Management.ResourceManager.Fluent.Core.<LoadPageWithWrapModelAsync>d__16.MoveNext() +376
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +60
   Microsoft.Azure.Management.ResourceManager.Fluent.Core.<LoadPage>d__15.MoveNext() +434
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +60
   Microsoft.Azure.Management.Monitor.Fluent.<ListEventDataAsync>d__10.MoveNext() +476
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +60
   Microsoft.Azure.Management.Monitor.Fluent.<ExecuteAsync>d__16.MoveNext() +456
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +60
   Microsoft.Azure.Management.Monitor.Fluent.<Microsoft-Azure-Management-Monitor-Fluent-IWithActivityLogsQueryExecute-ExecuteAsync>d__35.MoveNext() +198
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +60
   System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() +31
   .Monitoring.<QueryAzure>d__4.MoveNext() in AzureActivityLogService.cs:109
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +60
   System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() +31
   PppLearning.DevOps.MSAzure.Monitoring.<GetLogs>d__3.MoveNext() in AzureActivityLogService.cs:34
```

- Using v1.16.1 Fluent Nuget Packages
- ASP.net 4.7.2

**Comment by hovsepm** [comment]
@pppchris could you check if your `id` parameter is well formatted? If yes, please post here network payload (minus any sensitive data) of the request response including message headers.

**Comment by pppchris** [comment]
@hovsepm - found the problem. I am in AEST timezone. If I change the timestamp to be UTC now it works without an issue. 

`var log = await Repo.ActivityLogs
                .DefineQuery()
                .StartingFrom(DateTime.Now.AddDays(-1).ToUniversalTime())
                .EndsBefore(DateTime.Now.ToUniversalTime())
                .WithAllPropertiesInResponse()
                .FilterByResource(id)
               .ExecuteAsync();`

**Comment by hovsepm** [comment]
I will keep this issue open for some time until back-end service team will provide better error messages :)

**Comment by posypanka** [comment]
Thanks for that, it also did a trick for me ;-)

**Comment by liben-y** [comment]
I had the same issue from a GMT timezone. Changing to Universal time fixed the issue for me too, but not before I captured the call in Fiddler and made my own httpCall and converted the response json before finding this post.

I think the simplest solution to this would be for client sdk implementation to always convert to ToUniversaltime(). 

This way it'll work most of the time, specifically when the date passed in usually comes from DateTime.Now. 

Not sure what would happen if the kind is DateTimeKind.Unspecified, maybe throw an error in that case.

**Comment by yaohaizh** [comment]
@weidongxu-microsoft could you help take a look at this issue?

**Comment by weidongxu-microsoft** [comment]
@Dalubash

Would you confirm that the `/subscriptions/{subscriptionId}/providers/microsoft.insights/eventtypes/management/values` API only support UTC time in filter?
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2015-04-01/activityLogs_API.json#L39

SDK would adjust based on your feedback. Thanks.

PS current API behaviour:

Good
```
?$filter=eventTimestamp ge '2019-10-17T08:00:57.5909497Z' and eventTimestamp le '2019-10-24T08:00:57.5943305Z'
```

Bad
```
?$filter=eventTimestamp ge '2019-10-17T16:01:56.0770616+08:00' and eventTimestamp le '2019-10-24T16:01:56.0782560+08:00'
```

**Comment by weidongxu-microsoft** [comment]
Fixed in 1.28, to be release on Nov. 15.

Basically SDK would to ToUniversaltime(). DateTimeKind.Unspecified would go through default ToUniversaltime() behaviour.
